### PR TITLE
Update linter checklist with covered rule indications

### PR DESCRIPTION
**Pull request by vishrutshah**
Preview of the look can be found @ 

https://github.com/vishrutshah/azure-rest-api-specs/blob/update-linter-checklist/documentation/swagger-checklist.md

### CODE REVIEW

**Comment by kirthik** [comment]
Looks awesome 👏 Thanks for doing this! Can you squash commits to a single one?

**Comment by vishrutshah** [comment]
@kirthik Thanks. Admin who will merge this should be able to click on "Squash and Merge" while merging this PR.

**Comment by vishrutshah** [comment]
@BryanLa Could you please review this when you get a chance. Thanks!

File: documentation/swagger-checklist.md [hunk 1]
```diff
@@ -93,15 +94,15 @@ The Azure Developer Experience team will continue to deliver rich tools to valid
   - `ListByResourceGroup()` - list all resources in a resource group within a subscription.
   - `ListByParent()` - where "Parent" is a context specific suffix. It lists all resource under a parent.
 
-- [ ] **M1005**: Get operations *MUST* use the METHOD name "Get" or METHOD name start with "List"
+- [ ] **M1005**: :white_check_mark: Get operations *MUST* use the METHOD name "Get" or METHOD name start with "List"
```
Comment (BryanLa) [inline documentation/swagger-checklist.md]: `Get operations *MUST* use the METHOD name "Get" or METHOD name start with "List"` could be `Get operations *MUST* use the METHOD name "Get" or the METHOD name must start with "List"`

File: documentation/swagger-checklist.md [hunk 2]
```diff
@@ -122,11 +123,11 @@ The Azure Developer Experience team will continue to deliver rich tools to valid
...
+- [ ] **M2003**: :white_check_mark: If a format is specified along with a type, it *MUST* be from one of the supported formats - "int64", "int32", "float", "decimal", "double", "byte", "base64url", "unixtime", "date", "date-time", "duration", "date-time-rfc1123", "uuid", otherwise in codegeneration the artifact will be of the specified type and not as per the desired format. 
 
-- [ ] **M2004**: Long running (asynchronous) operations *MUST* be modeled as [long running operations](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/creating-swagger.md#longrunning) in Swagger.
+- [ ] **M2004**: :white_check_mark: Long running (asynchronous) operations *MUST* be modeled as [long running operations](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/creating-swagger.md#longrunning) in Swagger.
 
-- [ ] **M2005**: A longrunning operation *MUST* have a terminal success status code. At least 
+- [ ] **M2005**: :white_check_mark: A longrunning operation *MUST* have a terminal success status code. At least
```
Comment (BryanLa) [inline documentation/swagger-checklist.md]: `A longrunning operation` should be `A long-running operation`

**Review by BryanLa — COMMENTED** [review]
Hi @vishrutshah - done. I'd prefer @amarzavery have a look at this as well, for technical accuracy of Swagger properties. Also, at some point, will we change from "Swagger specification" to "OpenAPI Specification"?

**Comment by salameer** [comment]
Hi @BryanLa, please note that @amarzavery  is off the reviewing rotation is there any specific reason to pull him back in that the other reviewers cant help with?

File: documentation/swagger-checklist.md (hunk 1 above)
Comment (vishrutshah) [inline documentation/swagger-checklist.md]: Fixed

File: documentation/swagger-checklist.md (hunk 2 above)
Comment (vishrutshah) [inline documentation/swagger-checklist.md]: Fixed

**Review by BryanLa — APPROVED** [review]

**Comment by BryanLa** [comment]
@salameer - ah, OK, I was just hoping someone from the SDK team would provide review. If that's been done I can merge.

**Comment by vishrutshah** [comment]
@BryanLa Thanks for the review. I can "_Squash and merge_". Thanks!

**Pull request merged**